modules/freewheel-sspBidAdapter.js

```diff
@@ -306,7 +316,17 @@ export const spec = {
   */
   interpretResponse: function(serverResponse, request) {
     var bidrequest = request.bidRequest;
-    var playerSize = getBiggerSizeWithLimit(bidrequest.mediaTypes.banner.sizes, bidrequest.mediaTypes.banner.minSizeLimit, bidrequest.mediaTypes.banner.maxSizeLimit);
+    var playerSize = [];
+    if (bidrequest.sizes) {
+      // Backward compatible code, in case size still pass by sizes in bid request
+      playerSize = getBiggerSize(bidrequest.sizes);
+    } else if (bidrequest.mediaTypes.video && bidrequest.mediaTypes.video.playerSize) {
+      // If mediaTypes is video, get size from mediaTypes.video.playerSize per http://prebid.org/blog/pbjs-3
+      playerSize = bidrequest.mediaTypes.video.playerSize;
+    } else if (bidrequest.mediaTypes.banner.sizes) {
+      // If mediaTypes is banner, get size from mediaTypes.banner.sizes per http://prebid.org/blog/pbjs-3
+      playerSize = getBiggerSizeWithLimit(bidrequest.mediaTypes.banner.sizes, bidrequest.mediaTypes.banner.minSizeLimit, bidrequest.mediaTypes.banner.maxSizeLimit);

```

Comment:
Can you clarify where the `mediaTypes.banner.minSizeLimit` and `mediaTypes.banner.maxSizeLimit` fields get populated?  Are those extra fields meant to be added within the adUnit setup or something else?

Reply:
You are right. These two fields could be added within the adUnit set up together with the banner.sizes array. The idea is if any of the two limits (or both) is set, then only valid size within the [minSize -> maxSize] range could be used. If none of the limits are set, the default value of minSize [0, 0] and maxSize [Number.MAX_VALUE, Number.MAX_VALUE] will be set so that all the sizes in the array are treated as potentially valid.